pkg/build/controller/factory/factory.go

```diff
@@ -323,13 +323,35 @@ type podLW struct {
 
 // List lists all Pods that have a build label.
 func (lw *podLW) List() (runtime.Object, error) {
-	sel, _ := labels.Parse(buildapi.BuildLabel)
-	return lw.client.Pods(kapi.NamespaceAll).List(sel, fields.Everything())
+	// get builds with new label
+	sel, err := labels.Parse(buildapi.BuildLabel)
```

Comment:
refactor the logic to a function that podLW.List() and buildDeleteLW.List() can both call?